src/plots/cartesian/set_convert.js

```diff
@@ -67,24 +82,170 @@ module.exports = function setConvert(ax) {
         else return BADNUM;
     }
 
-    function fromLog(v) {
-        return Math.pow(10, v);
+    /*
+     * wrapped dateTime2ms that:
+     * - accepts ms numbers for backward compatibility
+     * - inserts a dummy arg so calendar is the 3rd arg (see notes below).
```

Comment:
yeah, it's a painful pattern for sure. A lot of trace types never see it though, as it's centralized in `ax.makeCalcdata` which gets the trace and data attribute as args already so it can look up the calendar. But if anyone can think of a better pattern I would be happy to refactor (after a bit of a break 😅  )

But - and this is perhaps an important point - the way I have it implemented traces do NOT inherit the axis calendar, so actually you never need to pay attention to `ax.calendar`, just the calendar that's specified in the trace itself. The only calendar inheritance is `layout.calendar` which sets the default for everything else. That's intended for what I expect will really be the most common use case of this, people really living by one of these calendars who want all their date data and all their date axes to use that calendar.